src/CommandingOfficers/Tech.java

```diff
@@ -0,0 +1,514 @@
+      while( !scoredSpaces.isEmpty() )
+      {
+        ScoredSpace ss = scoredSpaces.poll();
+        if( ss.score == s1.score )
+        {
+          equalSpaces.add(ss);
+          if( log ) System.out.println("           or " + ss.space.toString());
+        }
+        else break;
+      }
+
+      // Chose one at random if there are multiple equally-valid options.
+      int index = 0;
+      if( equalSpaces.size() > 1 )
+      {
+        Random rand = new Random(myCommander.units.size());
+        index = rand.nextInt(equalSpaces.size());
+      }
+      XYCoord landingZone = equalSpaces.get(index).space;
+      if(log) System.out.println("Landing at " + landingZone);
+
+      // Create our new unit and the teleport event to put it into place.
+      myCommander.units.add(techMech);
+      TeleportEvent techDrop = new TeleportEvent(gameMap, techMech, landingZone, TeleportEvent.AnimationStyle.DROP_IN, TeleportEvent.CollisionOutcome.KILL);
+      return techDrop;
+    }
+
+    private class ScoredSpace implements Comparable<ScoredSpace>
+    {
+      XYCoord space;
+      int score;
+
+      public ScoredSpace(XYCoord space, int score)
+      {
+        this.space = space;
+        this.score = score;
+      }
+      
+      @Override
+      public int compareTo(ScoredSpace other)
+      {
+        return other.score - score;
+      }
+    }
+  } // class TechDropAbility
+
+  /**
+   * Cobble together a new unit type from existing parts so that we don't have to modify
+   * the existing unit enum or damage tables. There are several downsides to this approach,
+   * like needing to use existing weapon/chassis types, and the fact that it'll break if
+   * we add a new UnitModelScheme with different units or weapons.
+   * @return A UnitModelScheme-compliant UnitModel to drop on enemy heads.
+   */
+  private UnitModel createBattleMechModel()
+  {
+    GameReadyModels grModels = gameRules.unitModelScheme.getGameReadyModels();
+    UnitModel mdTank = UnitModelScheme.getModelFromString("Md Tank", grModels.unitModels);
+    UnitModel BattleMech = mdTank.clone();
+    BattleMech.name = "BattleMech";
+    BattleMech.role = BattleMech.role | UnitModel.SURFACE_TO_AIR;
+//    BattleMech.moneyCost = 20000;
+    BattleMech.abilityPowerValue = 2.0;
+    BattleMech.maxFuel = 30;
+    BattleMech.maxAmmo = 10;
+    BattleMech.visionRange = 2;
+    BattleMech.movePower = 4;
+    BattleMech.propulsion = new FootMech();
+    BattleMech.healableHabs = new HashSet<TerrainType>(); // BattleMechs have specialized parts, not easy to repair.
+
+    UnitModel antiAir = UnitModelScheme.getModelFromString("Anti-Air", grModels.unitModels);
+    WeaponModel ratatat = antiAir.weapons.get(0).clone();
+    ArrayList<WeaponModel> weapons = new ArrayList<WeaponModel>();
+    weapons.add(BattleMech.weapons.get(0));
+    weapons.add(ratatat);
+    BattleMech.weapons = weapons;
+    return BattleMech;
+  }
```

Comment:
A few notes:
I would rather clone the CO's own models instead of pulling a whole extra `GameReadyModels` out.
Name matching is a poor choice, IMO. I would do a `getAllModels()` for `ASSAULT`, then pull out index 1 (or fall back to 0 if impossible). This fits with the documentation in `UnitModel`.
Likewise for the AA, I'd grab the first `SURFACE_TO_AIR` model.

It's pretty much otherwise how I'd do it.

As for improving the setup... I guess we could provide some sort of factory method on `UnitModelScheme`... but we'd run into the same sorts of difficulties setting up the "armor" and weapons.
This is definitely a drawback of abstracting the unit set out :P

Also, I see you have the `moneyCost` setter commented. It'd make sense to let you set it here, but that kind of messes with my idea of keeping the default cost "sacred" so as to avoid order-of-operations issues. You could accomplish this by setting the `moneyCostAdjustment` to the difference between your desired cost and the current cost, or we could go ahead and direct public access to `moneyCost`, I guess.

Reply:
> I would rather clone the CO's own models 

Sensible

> I would do a getAllModels() for ASSAULT, then pull out index 1

That depends on ordering to work right. If we added a unit type in between this would silently return the wrong thing. Using the name look up seems better precisely because it is more fragile. When it breaks, we'll know.

> moneyCost

Yeah, I thought about setting `moneyCostAdjustment`, but then if anyone else got BattleMechs, they'd be cheaper than Tech's and that feels wrong. That said, they aren't buildable or repairable, so it probably doesn't matter for now.

Reply:
> > I would do a getAllModels() for ASSAULT, then pull out index 1
> 
> That depends on ordering to work right. If we added a unit type in between this would silently return the wrong thing. Using the name look up seems better precisely because it is more fragile. When it breaks, we'll know.

The ordering being "sensible" for this sort of purpose is part of the contract of `UnitModelScheme`, as documented in the comments in `UnitModel`. If that needs to be more clear, we can enhance that documentation.
If I bring in a Wargroove unit set, there *won't* be a "Md Tank", but a Golem would serve just fine (and would be found via the process I outlined).

> Yeah, I thought about setting `moneyCostAdjustment`, but then if anyone else got BattleMechs, they'd be cheaper than Tech's and that feels wrong.

But... nobody else *can* get BattleMechs. They only exist for Tech, because she manufactures the model.
I guess Cinder could get a discount if she managed to buy one, but she'd have to steal the model first... and that's not a thing.
Anybody who doesn't tweak the cost adjustment would end up with the same price, since it'd come out of this very function with the correct adjustment in place.

> That said, they aren't ... repairable, so it probably doesn't matter for now.

Black Boats would like a word with you...
Also, cost matters for COP charge purposes.

Reply:
> Black Boats

True, forgot about those. I guess the adjustment field is as good as anything for now.

> Also, cost matters for COP charge purposes.

CO charge is separately determined by `abilityPowerValue`, isn't it?

Reply:
> CO charge is separately determined by `abilityPowerValue`, isn't it?

Planned feature. Not current feature.
Currently on `master`, it's AW2 charge (cost/9k for loss, half that for attack) + 0.1 star per HP you eliminate.